### PR TITLE
 Give local symbols reference equality.

### PR DESCRIPTION
**Pull request by olafurpg**
While upgrading to semanticdb v3.0 in scalafix we encountered that our
SemanticdbIndex was giving bogus results for local symbols. The bug was
not caught in our test suite, Guillaume discovered it while working on
SAM rewrite. This commit makes Symbol.Local use reference equality and
changes Database.load to use the same Symbol.Local instance for the same
id in the same document.

Applications that use alternative indexing techniques like SQL will need
to figure out another solution to distinguish between conflicting local
symbols between different files. One trick is to compute
definition/references on the fly from a Semanticdb for local symbols
because they cannot be referenced elsewhere.
1ad7790

### CODE REVIEW

**Comment by olafurpg** [comment]
Just noticed I'm not testing this fix properly, need to change the targeted test a bit.

**Comment by olafurpg** [comment]
I've fixed the implementation and tests to properly validate that this commit fixes the problem

File: langmeta/langmeta/shared/src/main/scala/org/langmeta/semanticdb/Symbol.scala [hunk 1]
```diff
@@ -19,6 +19,10 @@ object Symbol {
     override def toString = syntax
     override def syntax = id
     override def structure = s"""Symbol.Local("$id")"""
+    // Local symbols have reference equality because two local symbols from
+    // different source files can have the same id (for example, local0).
+    override def equals(obj: scala.Any): Boolean = this.eq(obj.asInstanceOf[AnyRef])
+    override def hashCode(): Int = System.identityHashCode(this)
```
Comment (xeno-by) [inline langmeta/langmeta/shared/src/main/scala/org/langmeta/semanticdb/Symbol.scala]: How about we just revert to the previous local symbol format? It is regrettable that with the current format we have to push identity management to schema users.

**Comment by olafurpg** [comment]
Reference equality is a bad idea. Let's not do this.

**Pull request closed without merging**